### PR TITLE
Add API integration tests with GitHub Actions CI (closes #38)

### LINKED ISSUE #38: Add API integration tests with GitHub Actions CI
There is no automated test coverage for the backend or TOTP service today. Adding API integration tests will catch regressions before they reach team prod and reduce manual QA going forward.

Scope (happy paths only), 12 tests across 4 files:
- Auth (2): register, login
- Credentials (5): full vault CRUD
- Internal (3): TOTP secret save and get, complete-mfa
- TOTP (2): setup, verify with a valid code

How it runs: GitHub Actions workflow `api-tests.yml` triggers on every push and pull request. Spins up a fresh MongoDB 7 container, starts the backend on port 3001 and TOTP service on port 4000, then runs `npm test` against the live services.

Out of scope:
- Edge cases (wrong password, lockout, validation, 401)
- Frontend tests
- Browser and cross-device testing

Verified working in my REPLICATE fork. All 12 tests pass in 32s end to end.

Does not affect existing deploys. The path filters on `backend.yml`, `frontend.yml`, and `totp.yml` correctly exclude `tests/**` and `api-tests.yml`, so merging the related PR will not redeploy any service.

### PR DESCRIPTION
**Pull request by perez-eduardo**
Adds API integration tests and a GitHub Actions workflow to run them on every push and pull request.

Files added:
- `tests/package.json`, jest, supertest, speakeasy
- `tests/helpers.js`, shared setup (registerUser, getJwt, getTotpSecret, generateTotpCode)
- `tests/auth.api.test.js`, 2 tests
- `tests/credentials.api.test.js`, 5 tests
- `tests/internal.api.test.js`, 3 tests
- `tests/totp.api.test.js`, 2 tests
- `.github/workflows/api-tests.yml`, the workflow

Total: 12 tests across 4 files.

Scope: happy paths only. Edge cases (wrong password, lockout, validation, 401) are intentionally out of scope for this PR.

How verified: tested in my REPLICATE fork. All 12 tests passed in 32s end to end. Workflow run: https://github.com/mmfclarke/secure-password-manager/actions/runs/25590849556

Does not affect existing deploys. Path filters on `backend.yml`, `frontend.yml`, and `totp.yml` correctly exclude `tests/**` and `api-tests.yml`, so merging this won't redeploy any service.

Closes #38

### CODE REVIEW

File: .github/workflows/api-tests.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        env:
+          MONGO_URI: mongodb://localhost:27017/securepm-test
+          JWT_SECRET: test-jwt-secret-for-ci-only
+          FRONTEND_ORIGIN: http://localhost:5173
+          TOTP_INTERNAL_SECRET: test-internal-secret-for-ci-only
+          PORT: 3001
+        run: |
+          npm start > /tmp/backend.log 2>&1 &
+          for i in {1..30}; do
+            if curl -sf http://localhost:3001/ > /dev/null; then
```
Comment (mmfclarke) [inline .github/workflows/api-tests.yml:45]: The backend readiness check is coupled to the application-level GET / route returning a 2xx response. This works right now because server.js has an explicit health route, but if that route is ever removed or renamed, this poll would silently time out on every CI run with no clear indication why. The TOTP check below uses nc -z localhost 4000 which just verifies the port is open regardless of what routes exist. It might be worth considering swapping this to nc -z localhost 3001 for the same robustness.

**Review by mmfclarke — APPROVED** [review]
Reviewed the workflow, all four test suites, and the helpers against the actual backend and TOTP service source code.

All endpoint paths, response shapes, and status codes match the implementation exactly. The getJwt helper using /complete-mfa to bypass TOTP is a good shortcut for credential tests. --runInBand correctly enforces serial execution for the ordered CRUD suite. The TOTP verify test generates a real speakeasy code against the actual stored secret, proper integration test, no mocking.

CI workflow spins up real MongoDB and both live services. All required env vars are provided and the backend startup validation passes cleanly. Left one inline note on the backend health check.

Nice work! Approving.

**Review by Thania97 — APPROVED** [review]
Reviewed the API integration tests and GitHub Actions workflow changes. The test suites are well organized and provide solid happy-path coverage for the authentication, credentials, internal API, and TOTP flows. The shared helper utilities also help keep the tests clean and reduce duplicated setup logic. 
The CI workflow was configured well and provides strong end-to-end integration testing with live MongoDB and both running services.
Everything looks solid overall! 

Approving!
